### PR TITLE
[ML] Missing chunkers for AzureOpenAIService, AzureAIStudioService and HuggingFace embeddings

### PR DESCRIPTION
**Pull request by davidkyle**
Adds chunking for AzureOpenAIService, AzureAIStudioService and the HuggingFace text embedding service. HuggingFace ELSER is not chunked.

### CODE REVIEW

**Comment by davidkyle** [comment]
@elasticmachine update branch

**Comment by davidkyle** [comment]
@elasticmachine update branch

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureaistudio/AzureAiStudioService.java [hunk 1]
```diff
@@ -105,23 +102,16 @@ protected void doChunkedInfer(
...
-        List<String> inputs,
-        InferenceServiceResults inferenceResults
-    ) {
-        if (inferenceResults instanceof InferenceTextEmbeddingFloatResults textEmbeddingResults) {
-            return InferenceChunkedTextEmbeddingFloatResults.listOf(inputs, textEmbeddingResults);
-        } else if (inferenceResults instanceof ErrorInferenceResults error) {
-            return List.of(new ErrorChunkedInferenceResults(error.getException()));
+        if (model instanceof AzureAiStudioModel baseAzureAiStudioModel) {
+            var actionCreator = new AzureAiStudioActionCreator(getSender(), getServiceComponents());
+            var batchedRequests = new EmbeddingRequestChunker(input, EMBEDDING_MAX_BATCH_SIZE, EmbeddingRequestChunker.EmbeddingType.FLOAT)
```
Comment (maxhniebergall) [inline x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureaistudio/AzureAiStudioService.java:107]: how do we know that the AzureAiStudioModel only uses FLOAT types?

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/huggingface/HuggingFaceBaseService.java [hunk 2]
```diff
@@ -36,14 +26,20 @@
...
 import static org.elasticsearch.xpack.inference.services.ServiceUtils.throwIfNotEmptyMap;
 
 public abstract class HuggingFaceBaseService extends SenderService {
 
+    /**
+     * The optimal batch size depends on the hardware the model is deployed on.
+     * For HuggingFace use a conservatively small max batch size as it is
+     * unknown how the model is deployed
+     */
+    static final int EMBEDDING_MAX_BATCH_SIZE = 20;
```
Comment (maxhniebergall) [inline x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/huggingface/HuggingFaceBaseService.java:41]: If the optimal batch size depends on the hardware, then I think we should make this user configurable. I think we can do that in a future PR if you want to get this one out now.

**Review by maxhniebergall — APPROVED** [review]
LGTM

File: x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureaistudio/AzureAiStudioService.java (hunk 1 above)
Comment (jonathan-buttner) [reply x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/azureaistudio/AzureAiStudioService.java:107]: Looks the response format is the same as openai which I believe supports `float` and `base64`. Elasticsearch only supports `float` and `byte` at the moment though.

**Review by jonathan-buttner — APPROVED** [review]

**Pull request merged**